include/deal.II/numerics/vector_tools.h

```diff
@@ -857,6 +857,47 @@ namespace VectorTools
                 const bool                          project_to_boundary_first = false);
 
   /**
+   * The same as above  for distributed Triangulations.
+   * The underlying MatrixFree implementation requires to specify the number
+   * of components and the degree of the used FiniteElement as additional
+   * template parameters.
+   * Currently, the implementation only allows for @p enforce_zero_boundary
+   * and @p project_to_boundary_first to be <tt>false</tt>.
+   * @p vec_result is assumed to not have a ghost elements.
+   */
+  template <int dim, typename VectorType, int spacedim, int components, int fe_degree>
```

Comment:
We could have the same discussion here as I had in the other PR about `components` and `fe_degree` being part of the public interface, which I'm not totally excited about.

If they need to be part of the public interface, maybe sort them to the front so that one only has to specify those template arguments that are not automatically deducible by the compiler?